File: Array/yangliu/Solution_06_11.java
public class Solution_06_11 {
    public ListNode oddEvenList(ListNode head) {
        if (head == null || head.next == null) {
            return head;
        }
        ListNode p=head;//记录奇数节点
        ListNode q=head.next;//记录偶数节点
        ListNode q1=head.next;
        while(p.next!=null&&q.next!=null){
            p.next=p.next.next;
            p=p.next;
            q.next=q.next.next;
            q=q.next;
        }
        p.next=q1;//最后一个奇数节点指向第一个偶数节点

        return head;
    }
    public static class ListNode {
        int val;
        ListNode next;

        ListNode(int x) {
            val = x;
            next = null;
        }
    }
}

File: Array/yangliu/Solution_06_11.c
#include <stdio.h>

struct ListNode *addEventList(struct ListNode *node);

struct ListNode {
    int val;
    struct ListNode *next;
};

int main() {
    printf("Hello, World!\n");
    struct ListNode *head;

    struct ListNode *listNode = addEventList(head);
    return 0;
}

struct ListNode *addEventList(struct ListNode *head) {
    if (head == NULL)return head;

    struct ListNode *o = head;//奇数
    struct ListNode *e = head->next;//偶数
    struct ListNode *e1 = head->next;//第一个偶数
    if (o->next!= NULL && e->next!= NULL) {
        o->next = o->next->next;
        o = o->next;
        e->next = e->next->next;
        e = e->next;
    }
    o->next = e1;//将最后一个奇数节点指向第一个偶数节点
    return head;
}


